doc/user-guide/data_model.rst

```diff
@@ -0,0 +1,251 @@
+.. _pyvista_data_model:
+
+
+PyVista Data Model
+==================
+PyVista uses the same data types as VTK, but structures them in a more
+pythonic manner for ease of use. If you'd like a background for how
+VTK structures its data, see `Introduction to VTK in Python by Kitware
+<https://vimeo.com/32232190>`_, as well as the numerous code examples
+on `Kitware's GitHub site
+<https://kitware.github.io/vtk-examples/site/>`_. An excellent
+introduction to mathematical concept implemented in VTK is provided by
+the `Discrete Differential Geometry YouTube Series
+<https://www.youtube.com/playlist?list=PL9_jI1bdZmz0hIrNCMQW1YmZysAiIYSSS>`_
+by Prof. Keenan Crane at Carnegie Melon.
+
+
+PyVista DataSet
+===============
+
+At the most fundamental level, all PyVista geometry objects are
+:ref:`ref_dataset`.  A dataset is a surface or volume in 3D space
+containing points, cells, and attributes describing that geometry.
+
+Geometry in PyVista is represented as points and cells.  In certain
+geometry types, such as :class:`pyvista.PolyData` or
+:class:`pyvista.UnstructuredGrids`, these cells must be explicitaly
+defined.  In other data types, such as :class:`pyvista.UniformGrid`,
+the cells are defined as a emergent property based on the shape of the
+point array.
+
+To see this in practice, let's create the simplest surface represented
+as a :class:`pyvista.PolyData`.  But first, we need to define our points.
+
+
+Points and Arrays in PyVista
```

Comment:
Also, in here, can you demo printing out a `pyvista` and/or `vtk` dataset and quickly showing what the information means? There's quite a lot more detail here than in your typical class `__repr__()`, which can be overwhelming to the newcomer.

e.g. If I make a `vtkImageData` dataset called `id` and print it out with `printi(id)`, I get this massive list like this (I think `pyvista` has shrunk this considerably, which is good, but it would do well to have a comparison between the printout of a raw `vtk` object and wrapped `pyvista` one:

```
vtkImageData (0000023932D73E80)
  Debug: Off
  Modified Time: 126
  Reference Count: 1
  Registered Events: (none)
  Information: 0000023934B751D0
  Data Released: False
  Global Release Data: Off
  UpdateTime: 0
  Field Data:
    Debug: Off
    Modified Time: 79
    Reference Count: 1
    Registered Events: (none)
    Number Of Arrays: 0
    Number Of Components: 0
    Number Of Tuples: 0
  Number Of Points: 250000
  Number Of Cells: 249001
  Cell Data:
    Debug: Off
    Modified Time: 82
    Reference Count: 1
    Registered Events:
      Registered Observers:
        vtkObserver (0000023932E88F50)
          Event: 33
          EventName: ModifiedEvent
          Command: 0000023934B75270
          Priority: 0
          Tag: 1
    Number Of Arrays: 0
    Number Of Components: 0
    Number Of Tuples: 0
    Copy Tuple Flags: ( 1 1 1 1 1 0 1 1 1 1 1 )
    Interpolate Flags: ( 1 1 1 1 1 0 0 1 1 1 1 )
    Pass Through Flags: ( 1 1 1 1 1 1 1 1 1 1 1 )
    Scalars: (none)
    Vectors: (none)
    Normals: (none)
    TCoords: (none)
    Tensors: (none)
    GlobalIds: (none)
    PedigreeIds: (none)
    EdgeFlag: (none)
    Tangents: (none)
    RationalWeights: (none)
    HigherOrderDegrees: (none)
  Point Data:
    Debug: Off
    Modified Time: 126
    Reference Count: 1
    Registered Events:
      Registered Observers:
        vtkObserver (0000023932E89370)
          Event: 33
          EventName: ModifiedEvent
          Command: 0000023934B75270
          Priority: 0
          Tag: 1
    Number Of Arrays: 1
    Array 0 name = my array
    Number Of Components: 1
    Number Of Tuples: 250000
    Copy Tuple Flags: ( 1 1 1 1 1 0 1 1 1 1 1 )
    Interpolate Flags: ( 1 1 1 1 1 0 0 1 1 1 1 )
    Pass Through Flags: ( 1 1 1 1 1 1 1 1 1 1 1 )
    Scalars:
      Debug: Off
      Modified Time: 76
      Reference Count: 2
      Registered Events: (none)
      Name: my array
      Data type: double
      Size: 250000
      MaxId: 249999
      NumberOfComponents: 1
      Information: 0000000000000000
      Name: my array
      Number Of Components: 1
      Number Of Tuples: 250000
      Size: 250000
      MaxId: 249999
      LookupTable: (none)
    Vectors: (none)
    Normals: (none)
    TCoords: (none)
    Tensors: (none)
    GlobalIds: (none)
    PedigreeIds: (none)
    EdgeFlag: (none)
    Tangents: (none)
    RationalWeights: (none)
    HigherOrderDegrees: (none)
  Bounds:
    Xmin,Xmax: (0, 499)
    Ymin,Ymax: (0, 499)
    Zmin,Zmax: (0, 0)
  Compute Time: 127
  Spacing: (1, 1, 1)
  Origin: (0, 0, 0)
  Direction: (1, 0, 0, 0, 1, 0, 0, 0, 1)
  Dimensions: (500, 500, 1)
  Increments: (0, 0, 0)
  Extent: (0, 499, 0, 499, 0, 0)
```

Reply:
@adamgranthendry just a small correction: it's usually `__str__` that is verbose and `__repr__` is short and barely informative in VTK/PyVista. It took me a pretty long time to find the verbose output, because I always rely on evaluation in the REPL, which uses `__repr__` :)

Reply:
@adeak Ah, excellent point! Yes, thanks for that!